### DISCUSSION TITLE
Blocking other apps

### DISCUSSION THREAD

**Discussion by dohanin** [Waboodoo/HTTP-Shortcuts#350] [Q&A] [answered]
I am using tasker heavily and one problem I encounter often is the freezing due to HTTP requests.  Because the HTTP Request action has to wait for the request to complete or time out before proceeding with the next action, it often bottles up all my pending requests.  So, I have been looking for any alternatives to run HTTP requests in parallel and found HTTP-Shortcuts.

It looks perfect to me.  However, HTTP-Shortcuts would show a waiting animation if the request takes some time to complete (e.g. slow network), and it would block screen.  Even if I disabled the permission of drawing over other apps, it still shows.

Is there anyway to run HTTP-Shortcuts completely silently in background without showing any animations?

Many thanks.

**Comment by Waboodoo** [comment, marked as answer]
There is a possibility, yes. It's currently a hidden/experimental feature and therefore not really documented, but it might work for your usecase. In order for a shortcut to run without the blocking loading indicator, the following conditions need to be fulfilled:
- The shortcut must not make use of file uploads, neither for the request body or for request parameters
- On the shortcut's "Response Handling" screen, either both the "On Success" and "on Failure" dropdown menues need to be set to "Show nothing (run silently)" or the "Display Type" option needs to be set to "Toast Popup" and the app needs to be granted the Notification permission (there's no UI for this inside the app, you need to do it through the app settings menu)
- The "Store response into file" option on the "Response Handling" screen must not be enabled
- On the shortcut's "Scripting" screen, there must not be any code in the "Run on Success" and "Run of Failure" text fields
- On the "Trigger & Execution Settings" screen, the "Wait for connection when offline" option must not be enabled
- The total size of all used variables (name + values) in the shortcut must be below the threshold of 8000 characters
- Battery & Data Saver modes need to be displayed or the app exempt from them

Furthermore, it might also be necessary to enable the "Draw over other apps" permission, as the Tasker integration sometimes doesn't work properly without it.

If all of the above conditions are met, the HTTP Shortcuts app will try to run your shortcut in a sort-of "headless" or "UI-less" mode. When used with Tasker, Tasker will not wait for the response and continue immediately.

As this feature is not fully fleshed out and still subject to changes (I'm hoping to soon make this much more useful and discoverable) I currently can't promise that this will solve your issues, but it might.
